lib/ex_unit/lib/ex_unit/case.ex

```diff
@@ -316,7 +319,19 @@ defmodule ExUnit.Case do
       Module.put_attribute(mod, :ex_unit_test_names, Map.put(test_names, name, true))
     end
 
-    Module.delete_attribute(mod, :tag)
+    Enum.each [:tag | registered_attributes], fn(attribute) ->
+      Module.delete_attribute(mod, attribute)
+    end
+  end
+
+  @doc """
+  Registers a new attribute to be used during `ExUnit.Case` tests.
```

Comment:
The docs could be more detailed. It could mention that the attributes are automatically cleaned up after every test is defined. We also need to update the `@moduledoc` to mention `@registered` is one of the private module attributes that cannot be changed.